diff -r 8c3f29ff2ba4 dot.vim/dist/vim-fakeclip/autoload/fakeclip.vim
--- a/dot.vim/dist/vim-fakeclip/autoload/fakeclip.vim	Sun Apr 24 23:59:00 2011 +0900
+++ b/dot.vim/dist/vim-fakeclip/autoload/fakeclip.vim	Mon Apr 25 00:00:25 2011 +0900
@@ -34,7 +34,7 @@
 endif
 
 
-if executable('screen')
+if executable('screen') && $STY != ''
   let s:TERMINAL_MULTIPLEXER_TYPE = 'gnuscreen'
 elseif executable('tmux') && $TMUX != ''
   let s:TERMINAL_MULTIPLEXER_TYPE = 'tmux'
