diff -r 7e525b02f462 dot.vim/dist/vimwiki/autoload/vimwiki_html.vim
--- a/dot.vim/dist/vimwiki/autoload/vimwiki_html.vim	Thu Feb 03 12:04:27 2011 +0900
+++ b/dot.vim/dist/vimwiki/autoload/vimwiki_html.vim	Thu Feb 03 12:18:26 2011 +0900
@@ -1264,6 +1264,25 @@
     call s:close_tag_table(state.table, lines)
     call extend(ldest, lines)
 
+    let new_ldest = []
+    let para = 0
+    for line in ldest
+      if para
+        let para_line = remove(new_ldest, -1)
+        let sp = para_line[-1] == matchstr(para_line, ".$") || line[0] == matchstr(line, "^.") ? ' ' : ''
+        call add(new_ldest, para_line . sp . line)
+      else
+        call add(new_ldest, line)
+      endif
+      if line =~ '^<p>'
+        let para = 1
+      endif
+      if line =~ '^</p>'
+        let para = 0
+      endif
+    endfor
+    let ldest = new_ldest
+
     let title = s:process_title(placeholders, fnamemodify(a:wikifile, ":t:r"))
     call extend(ldest, s:get_html_header(title, subdir, &fileencoding), 0)
     call extend(ldest, s:get_html_footer())
